### PR TITLE
Add TimeoutOption to serial executor

### PR DESCRIPTION
**Pull request by kl52752**

### CODE REVIEW

File: pkg/cloud/rgraph/exec/executor_serial.go [hunk 1]
```diff
@@ -63,7 +63,22 @@ type serialExecutor struct {
 
 var _ Executor = (*serialExecutor)(nil)
 
+// Run executes pending actions sequentially.
+//
+// Use TimeoutOption to define timeout for executor to launch new actions.
+// Note that when timeout occurs the executor will block until active action
```
[the remark this answers is not in the document]
Comment (kl52752) [reply pkg/cloud/rgraph/exec/executor_serial.go:69]: This line `the executor will block until active action` is specific for serial executor only. In parallel we are waiting for orphans and actions are run in go routines so we can easily abandon them. In this case we are not running actions in go routines that means that even when timeout occurs we will not stop active operation, we need to wait until it finished.

[the remark this answers is not in the document]
Comment (bowei) [reply pkg/cloud/rgraph/exec/executor_serial.go:69]: Timeout should result in the context being canceled. Doesn't this cause any properly managed running contexts to also return immediately?

**Pull request merged**